site/cypress/e2e/Cypress-Gradeable/office_hours_queue.spec.js

```diff
@@ -0,0 +1,145 @@
+    cy.get('[data-testid="toggle-filter-settings"]').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('[data-testid="delete-queue-btn"]').last().click();
+};
+const disableQueue = () => {
+    cy.visit(['sample', 'config']);
+    cy.get('[data-testid="queue-enabled"]').should('be.checked');
+    cy.get('[data-testid="queue-enabled"]').check();
+};
+const openNewQueue = (queueName, queueCode='') => {
+    cy.get('#nav-sidebar-queue').click();
+    cy.get('[data-testid="toggle-new-queue"]').should('exist').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('[data-testid="new-queue-code"]').type(queueName);
+    if (queueCode.length > 0) {
+        cy.get('[data-testid="new-queue-token"]').type(queueCode);
+    }
+    else {
+        cy.get('[data-testid="new-queue-rand-token"]').click();
+    }
+    cy.get('[data-testid="open-new-queue-btn"]').click();
+};
+
+const changeQueueCode = (queueName, queueCode='') => {
+    cy.get('[data-testid="toggle-filter-settings"]').should('exist').click();
+    cy.get('[data-testid="popup-window"]').should('exist');
+    cy.get('[data-testid="old-queue-code"]').select(queueName);
+    if (queueCode.length > 0) {
+        cy.get('[data-testid="old-queue-token"]').type(queueCode);
+    }
+    else {
+        cy.get('[data-testid="old-queue-rand-token"]').click(); // random code
+    }
+    cy.get('[data-testid="change-code-btn"]').click(); // update it
+};
+const switchUser = (account) => {
+    cy.logout();
+    cy.login(account);
+    cy.visit(['sample', 'office_hours_queue']);
+};
+const studentJoinQueue = (queueName, queueCode) => {
+    cy.get('[data-testid="queue-code"]').select(queueName).invoke('val'); // in which queue you want to join
+    cy.get('[data-testid="queue-code"]').should('contain', queueName);
+    cy.get('#token-box').type(queueCode);
+    cy.get('[data-testid="join-queue-btn"]').click();
+};
+const editAnnouncement = (text='') => {
+    // openAnnouncementSettings
+    cy.get('[data-testid="toggle-announcement-settings"]').click();
+    cy.get('#announcement-settings').should('exist');
+    cy.get('#queue-announcement-message').clear();
+    if (text.length > 0) {
+        cy.get('#queue-announcement-message').type(text);
+    }
+    // saveAnnouncementSettings
+    cy.get('[data-testid="save_announcement"]').click();
+};
+describe('test office hours queue', () => {
+    it('opened new queue, students joining queue, started and finished helping student', () => {
+        cy.login();
+        enableQueue();
+        // deleting the Lab help and homework debugging
+        deleteQueue();
+        deleteQueue();
+        openNewQueue(queueName, queueCode);
+        // cy.get('.alert-success').contains('New queue added');
+        openNewQueue(queueName, queueCode1); // same name but used different code
+        cy.get('.alert-error').contains('Unable to add queue. Make sure you have a unique queue name');
```

Comment:
If you are asserting that the error should contain the text, use .should('contain', 'message'). Apply throughout.

Reply:
done sir